app/src/main/java/com/example/lab3/MainActivity.java

```diff
@@ -0,0 +1,125 @@
+        setContentView(R.layout.activity_main);
+        bindServices();
+
+        Button startServiceButton = findViewById(R.id.start_service_button);
+        Button endServiceButton = findViewById(R.id.end_service_button);
+        Button getCurrentDataButton = findViewById(R.id.get_current_data_button);
+        Button getMoneyButton = findViewById(R.id.get_money_button);
+
+        IntentFilter filter = new IntentFilter(MyIntentService.class.getName());
+        LocalBroadcastManager.getInstance(this).registerReceiver(myBroadcastReceiver, filter);
+
+        startServiceButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent = new Intent(MainActivity.this, MyStartedService.class);
+                startService(intent);
+            }
+        });
+
+        endServiceButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent = new Intent(MainActivity.this, MyStartedService.class);
+                stopService(intent);
+            }
+        });
+
+        getCurrentDataButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Toast toast = Toast.makeText(MainActivity.this, myBoundService.getCurrentDate(), Toast.LENGTH_LONG);
+                toast.show();
+            }
+        });
+
+        getMoneyButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                final String ACTION_GIVE_MONEY = "GIVE_MONEY";
+                final String EXTRA_MIN_VALUE = "MIN_VALUE";
+                final String EXTRA_MAX_VALUE = "MAX_VALUE";
+
+                Intent intent = new Intent(MainActivity.this, MyIntentService.class);
+                intent.setAction(ACTION_GIVE_MONEY);
+                intent.putExtra(EXTRA_MIN_VALUE, 100);
+                intent.putExtra(EXTRA_MAX_VALUE, 1000);
+                startService(intent);
+            }
+        });
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        unbindServices();
+        LocalBroadcastManager.getInstance(this).unregisterReceiver(myBroadcastReceiver);
+    }
+
+    private void bindServices() {
+        bindServicesHelper(myStartedService, myStartedServiceConnection, MyStartedService.class);
+        bindServicesHelper(myBoundService, myBoundServiceConnection, MyBoundService.class);
+
+    }
+
+    private void bindServicesHelper(Service service, ServiceConnection serviceConnection, Class classInstance) {
+        if (service == null) {
+            Intent intent = new Intent(this, classInstance);
+            bindService(intent, serviceConnection, BIND_AUTO_CREATE);
+        }
+    }
+
+    private void unbindServices() {
+        unbindService(myStartedServiceConnection);
```

Comment:
The service for the first task is not a bound one, it's just started, no need for a service connection for it and for bind/unbind.